_extras/snippets_library/UCL_sge/14/matlab_example.snip

```diff
@@ -0,0 +1,138 @@
+{{ site.host_prompt }} module list
+```
+{: .bash}
+```
+No Modulefiles Currently Loaded.
+```
+{: .output}
+```
+{{ site.host_prompt }} module load matlab/full/r2017a/9.2
+```
+{: .bash}
+```
+matlab/full/r2017a/9.2(96):ERROR:151: Module 'matlab/full/r2017a/9.2' depends on one of the module(s) 'gcc-libs/4.9.2'
+matlab/full/r2017a/9.2(96):ERROR:102: Tcl command execution failed: prereq gcc-libs
+```
+{: .output}
+```
+{{ site.host_prompt }} module load gcc-libs/4.9.2
+{{ site.host_prompt }} module load matlab/full/r2017a/9.2
+```
+{: .bash}
+```
+matlab/full/r2017a/9.2(97):ERROR:151: Module 'matlab/full/r2017a/9.2' depends on one of the module(s) 'xorg-utils/X11R7.7'
+matlab/full/r2017a/9.2(97):ERROR:102: Tcl command execution failed: prereq xorg-utils/X11R7.7
+```
+{: .output}
+```
+{{ site.host_prompt }} module load xorg-utils/X11R7.7
+{{ site.host_prompt }} module load matlab/full/r2017a/9.2
+```
+{: .bash}
+```
+~/.matlab is a symbolic link pointing to /home/cceagil/Scratch/.matlab
+
+Matlab setup complete type matlab to start Matlab.
+```
+{: .output}
+
+```
+{{ site.host_prompt }} matlab -nodisplay -nosplash -nodesktop
+```
+
+{: .bash}
+```
+                                               < M A T L A B (R) >
+                                     Copyright 1984-2017 The MathWorks, Inc.
+                                      R2017a (9.2.0.556344) 64-bit (glnxa64)
+                                                  March 27, 2017
+
+ 
+To get started, type one of these: helpwin, helpdesk, or demo.
+For product information, visit www.mathworks.com.
+ 
+>> quit
+```
+{: .output}
+
+Note that you cannot load two different versions of the same software at once.
+Currently, we have loaded `matlab/full/r2017a/9.2`. Let's try also loading
+`matlab/full/r2015b/8.6`:
+
+```
+{{ site.host_prompt }} module load matlab/full/r2015b/8.6
+```
+{: .bash}
+```
+matlab/full/r2015b/8.6(108):ERROR:150: Module 'matlab/full/r2015b/8.6' conflicts with the currently loaded module(s) 'matlab/full/r2017a/9.2'
+matlab/full/r2015b/8.6(108):ERROR:102: Tcl command execution failed: conflict matlab
+```
+{: .output}
+
+As we can see, we get an error message about conflicts.
+If we do indeed wish to load version r2015b/8.6, we can say
+```
+{{ site.host_prompt }} module unload matlab/full/r2017a/9.2
+{{ site.host_prompt }} module load matlab/full/r2015b/8.6
+```
+or, in one step:
+```
+{{ site.host_prompt }} module swap matlab matlab/full/r2015b/8.6
+```
+```
+{{ site.host_prompt }} module list
+```
+{: .bash}
+```
+Currently Loaded Modulefiles:
+  1) gcc-libs/4.9.2                 9) gerun                         17) userscripts/1.4.0
+  ...                              ...                               ...
+  6) apr-util/1.5.4                14) emacs/24.5                    22) xorg-utils/X11R7.7
+  7) subversion/1.8.13             15) tmux/2.2                      23) matlab/full/r2015b/8.6
+  8) screen/4.2.1                  16) mrxvt/0.5.4
+```
+{: .output}
```

Comment:
Very good example! I wish we could avoid the dependency madness...

Reply:
Thanks.